lib/action_view/component.rb

```diff
@@ -67,7 +67,9 @@ def inherited(child)
       # We could in theory do this on app boot, at least in production environments.
       # Right now this just compiles the template the first time the component is rendered.
       def compile
-        return if @compiled
+        # Only use compiled method if we're in production. Eventually, we'll probably want to
+        # use something like Rails.application.config for this toggle instead.
+        return if @compiled && Rails.env.production?
```

Comment:
Best to avoid explicitly checking the `Rails.env` value (which is supposed to have a purely user-defined meaning).

I think `ActionView::Base.cache_template_loading` would be a good fit here.